tests/spread/general/health-checks/rockcraft.yaml

```diff
@@ -0,0 +1,61 @@
+name: healthy-rock
+summary: A ROCK with health checks
+description: A test ROCK with a bunch of different Pebble checks
+license: Apache-2.0
+
+version: latest
+base: "ubuntu:22.04"
+
+services: 
+  webserver:
+    override: replace
+    command: timeout 30 nginx -g 'daemon off;'
+    environment:
+      TZ: UTC
+    startup: enabled
+    on-failure: shutdown
+
+checks:
+  http-check:
+    override: replace
+    level: alive
+    period: 1s
+    http:
+      url: http://localhost/
+  tcp-check:
+    override: replace
+    level: ready
+    tcp:
+      port: 80
+  exec-check:
+    override: replace
+    exec:
+      command: pebble services
+
+platforms:
+  amd64:
+
+package-repositories:
+  - type: apt
+    url: https://nginx.org/packages/mainline/ubuntu
+    key-id: 573BFD6B3D8FBC641079A6ABABF5BD827BD9BF62
+    suites: 
+      - jammy
+    components:
+      - nginx
+    priority: always
```

Comment:
Out of curiosity: why did you add this repository rather than using the nginx package in jammy main? 

(I'm guessing there are either packaging differences, something else you're trying to test, or that you're just more comfortable with that package. All three are totally valid, I'm just curious.)

Reply:
this is to mimic the upstream build recipe, from nginx -> https://github.com/nginxinc/docker-nginx/blob/1a8d87b69760693a8e33cd8a9e0c2e5f0e8b0e3c/stable/debian/Dockerfile.

iirc, the way of installing it via the standard jammy archives is slightly different, so this was just easier to inherit from our existing tutorials